ql/patterns/observable.hpp

```diff
@@ -264,14 +264,26 @@ namespace QuantLib {
 #include <boost/enable_shared_from_this.hpp>
 #include <set>
 
+
+
 namespace QuantLib {
+    namespace ext {
+
+        #if defined(QL_USE_STD_SHARED_PTR)
+        using std::enable_shared_from_this;
```

Comment:
I would move this (and the inclusion of `<boost/enable_shared_from_this.hpp>` above) to `ql/shared_ptr.hpp` where the other related classes are added to namespace `ext`.